public/app.js

```diff
@@ -0,0 +1,101 @@
+var app = angular.module('libraryApp', ['ngRoute', 'ngResource']);
+
+////////////
+// ROUTES //
+////////////
+
+app.config(['$routeProvider', '$locationProvider', function ($routeProvider, $locationProvider) {
+  $routeProvider
+    .when('/', {
+      templateUrl: 'templates/books/index.html',
+      controller: 'BooksIndexCtrl'
+    })
+    .when('/books/:id', {
+      templateUrl: 'templates/books/show.html',
+      controller: 'BooksShowCtrl'
+    });
+    $locationProvider
+    .html5Mode({
+      enabled: true,
+      requireBase: false
+    });
+}]);
+
+/////////////////
+// API CALL //
+/////////////////
+
+app.factory("Book", ["$resource", function($resource) {
+  return $resource('https://api.parse.com/1/classes/Book/:bookId', {bookId: "@bookId"}, {
+    query: {
+      method: 'GET',
+      headers: {
+      'X-Parse-Application-Id': 'JjfvfpGDvmx9etMgv916ittvni3QkdvdXY6ODbEc',
```

Comment:
I applaud your unique anti-authority approach to reducing code duplication and legible indentation. That said, this amount of ⌘+ c / ⌘ + v isn't what I'd recommend.